src/app/tests/TestReadInteraction.cpp

```diff
@@ -242,6 +295,133 @@ class MockInteractionModelApp : public chip::app::ReadClient::Callback
         }
     }
 
+    // Log the current captures in a code-like format, to more easily update tests
+    void LogCaptures(const char * heading)
+    {
+        ChipLogProgress(Test, "Captured attributes (%s):", heading);
+
+        for (unsigned i = 0; i < mReceivedAttributePaths.size(); i++)
+        {
+            const auto & path = mReceivedAttributePaths[i];
+            chip::StringBuilder<128> argsBuffer;
+
+            if (path.mEndpointId >= 0xff)
+            {
+                argsBuffer.AddFormat("0x%X, ", path.mEndpointId);
+            }
+            else
+            {
+                argsBuffer.AddFormat("%u, ", path.mEndpointId);
+            }
+
+            if (path.mClusterId >= 0xff)
+            {
+                argsBuffer.AddFormat("0x%lX, ", static_cast<unsigned long>(path.mClusterId));
+            }
+            else
+            {
+                argsBuffer.AddFormat("%lu, ", static_cast<unsigned long>(path.mClusterId));
+            }
+
+            switch (path.mAttributeId)
+            {
+            case ClusterRevision::Id:
+                argsBuffer.Add("ClusterRevision::Id");
+                break;
+            case FeatureMap::Id:
+                argsBuffer.Add("FeatureMap::Id");
+                break;
+            case GeneratedCommandList::Id:
+                argsBuffer.Add("GeneratedCommandList::Id");
+                break;
+            case AcceptedCommandList::Id:
+                argsBuffer.Add("AcceptedCommandList::Id");
+                break;
+            case AttributeList::Id:
+                argsBuffer.Add("AttributeList::Id");
+                break;
+            default:
+                if (path.mAttributeId >= 0xff)
+                {
+                    argsBuffer.AddFormat("0x%lX", static_cast<unsigned long>(path.mAttributeId));
+                }
+                else
+                {
+                    argsBuffer.AddFormat("%lu", static_cast<unsigned long>(path.mAttributeId));
+                }
+                break;
+            }
+
+            auto listSize = mReceivedListSizes[i];
+            if (listSize.has_value())
+            {
+                argsBuffer.AddFormat(", /* listSize = */ %u", listSize.value());
+            }
+
+            ChipLogProgress(Test, "  AttributeCaptureAssertion(%s),", argsBuffer.c_str());
+        }
+        // TODO
```

Comment:
What is this TODO for?

Reply:
Ooops ... remaining for before I had written this function. Thank you for noticing!